src/engine/cache_engine.h

```diff
@@ -32,10 +32,11 @@ typedef enum {
 	ocf_req_cache_mode_max,
 } ocf_req_cache_mode_t;
 
-struct ocf_io_if {
-	int (*read)(struct ocf_request *req);
+typedef int (*ocf_engine_cb)(struct ocf_request *req);
 
-	int (*write)(struct ocf_request *req);
+struct ocf_io_if {
+	ocf_engine_cb read;
+	ocf_engine_cb write;
```

Comment:
Maybe if we're here optimizing, we could do something like
```c
struct ocf_io_if {
        union {
                struct {
                        ocf_engine_cb read;
                        ocf_engine_cb write;
                };
                ocf_engine_cb cbs[2];
};
``` 
(are we C11 [yet](https://news.itsfoss.com/linux-kernel-5-18-release/)? :))
and then change all ternary operator to just 
```c
IO_IFS[io_if_type].cbs[OCF_READ]
```
?
Didn'd do any benchmarking, but I think it's a quite cheap way to eliminate a couple of branches in I/O path.